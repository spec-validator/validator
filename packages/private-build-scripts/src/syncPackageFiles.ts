import glob from 'glob'
import fs from 'fs'
import util from 'util'

import { Task } from 'just-task'

import { removePrefix, flatMap } from '@spec-validator/utils/utils'

import copyRec from './copyRec'

import { read } from './readAndWrite'

const copyFile = util.promisify(fs.copyFile)

export default (projectPath: string): Task => async () => {
  const packageJson: Record<string, any>  = read(`${projectPath}/package.json`)

  const files: string[] = packageJson.files || []

  copyFile(`${projectPath}/README.md`, `${projectPath}/dist/README.md`)

  flatMap(
    files.map(it => glob.sync(`${projectPath}/${it}`)),
  ).forEach(src => {
    const relPath = removePrefix(src, `${projectPath}/`)
    const dest = `${projectPath}/dist/${relPath}`
    copyRec(src, dest)
  })
}
