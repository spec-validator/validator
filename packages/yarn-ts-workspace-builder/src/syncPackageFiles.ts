import { sync } from 'glob'
import { copyFileSync } from 'fs'

import { removePrefix, flatMap } from '@spec-validator/utils/utils'

import copyRec from './copyRec'

import { read } from './readAndWrite'
import { DIST } from './generateTsConfigJson'

export default (projectPath: string): void => {
  copyFileSync(`${projectPath}/README.md`, `${projectPath}/${DIST}/README.md`)

  const packageJson: Record<string, any>  = read(`${projectPath}/package.json`)

  const files: string[] = packageJson.files || []

  flatMap(
    files.map(it => sync(`${projectPath}/${it}`)),
  ).forEach(src => {
    const relPath = removePrefix(src, `${projectPath}/`)
    const dest = `${projectPath}/${DIST}/${relPath}`
    copyRec(src, dest)
  })
}
