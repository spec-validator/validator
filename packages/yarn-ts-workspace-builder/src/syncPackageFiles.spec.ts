import syncPackageFiles from './syncPackageFiles'

const copyFileSync = jest.fn()

jest.mock('fs', () => ({
  copyFileSync: (src: string, trg: string) => copyFileSync(src, trg),
  __esModule: true,
}))

jest.mock('./readAndWrite', () => ({
  read: () => ({
    files: ['file1', 'file2'],
  }),
  __esModule: true,
}))

jest.mock('glob', () => ({
  sync: (path: string) => [path],
  __esModule: true,
}))

const copyRec = jest.fn()

jest.mock('./copyRec', () => ({
  default: (src: string, trg: string) => copyRec(src, trg),
  __esModule: true,
}))

test('sync', () => {
  syncPackageFiles('./project')
  expect(copyFileSync.mock.calls).toMatchSnapshot()
  expect(copyRec.mock.calls).toMatchSnapshot()
})
