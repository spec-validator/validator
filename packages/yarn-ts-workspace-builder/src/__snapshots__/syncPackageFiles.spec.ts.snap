// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sync 1`] = `
Array [
  Array [
    "./project/README.md",
    "./project/dist/README.md",
  ],
]
`;

exports[`sync 2`] = `
Array [
  Array [
    "./project/file1",
    "./project/dist/file1",
  ],
  Array [
    "./project/file2",
    "./project/dist/file2",
  ],
]
`;
